%% BioMed_Central_Tex_Template_v1.06
%%                                      %
%  bmc_article.tex            ver: 1.06 %
%                                       %

%%IMPORTANT: do not delete the first line of this template

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                     %%
%%  LaTeX template for BioVis 2014  %%
%%      article submissions     %%
%%          adapted from BMC    %%
%%          <8 Jan 2014>              %%
%% Liz Marai (g.elisabeta.marai@gmail.com) %%
%%                                     %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

	
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                                                 %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% additional documentclass options:
%  [doublespacing]
%  [linenumbers]   - put the line numbers on margins

%%% loading packages, author definitions

\documentclass[twocolumn]{bmcart}% uncomment this for twocolumn layout 



%%% Load packages
%\usepackage{amsthm,amsmath}
%\RequirePackage{natbib}
%\RequirePackage{hyperref}
\usepackage[utf8]{inputenc} %unicode support
%\usepackage[applemac]{inputenc} %applemac support if unicode package fails
%\usepackage[latin1]{inputenc} %UNIX support if unicode package fails
\usepackage{graphicx}
\usepackage{color}
\usepackage[textsize=small,colorinlistoftodos]{todonotes}
\usepackage{algorithm}
\usepackage{algorithmicx}
\usepackage{algpseudocode}
\usepackage{xspace}
\usepackage{subfigure}
\usepackage{amsfonts}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                             %%
%%  If you wish to display your graphics for   %%
%%  your own use using includegraphic or       %%
%%  includegraphics, then comment out the      %%
%%  following two lines of code.               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


%\def\includegraphic{}
%\def\includegraphics{}



%%% Put your definitions there:
\startlocaldefs
\endlocaldefs


% Defines usefull commands.
\newcommand{\ie}{i.e.,~}
\newcommand{\eg}{e.g.,~}

\algrenewcommand\algorithmicrequire{\textbf{Input:}}
\algrenewcommand\algorithmicensure{\textbf{Output:}}
\algnewcommand\And{\textbf{and}\xspace}
\algnewcommand\Or{\textbf{or}\xspace}

\newcommand{\angstrom}{\mbox{\normalfont\AA}}

%%% Begin ...
\begin{document}

%%% Start of article front matter
\begin{frontmatter}

\begin{fmbox}
\dochead{Research}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the title of your article here     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{Interactive Exploration of Ligand Transportation through Protein Tunnels}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Do not enter the authors here for        %%
%%  a double-blind review. Otherwise        %%
%% specify information, if available,       %%
%% in the form:                             %%
%%   <key>={<id1>,<id2>}                    %%
%%   <key>=                                 %%
%% Comment or delete the keys which are     %%
%% not used. Repeat \author command as much %%
%% as required.                             %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\author[
   addressref={aff1},
   email={furmanova@mail.muni.cz}
]{\inits{KF}\fnm{Katar\'{i}na} \snm{Furmanov\'{a}}}
\author[
   addressref={aff1},                   % id's of addresses, e.g. {aff1,aff2}
   email={jaresova@mail.muni.cz}   % email address
]{\inits{MJ}\fnm{Miroslava} \snm{Jare\v{s}ov\'{a}}}
\author[
   addressref={aff1,aff2},                   % id's of addresses, e.g. {aff1,aff2}
   email={xbyska@fi.muni.cz}   % email address
]{\inits{JB}\fnm{Jan} \snm{By\v{s}ka}}
\author[
   addressref={aff1},                   % id's of addresses, e.g. {aff1,aff2}
   email={xjurc@fi.muni.cz}   % email address
]{\inits{AJ}\fnm{Adam} \snm{Jur\v{c}\'{i}k}}
\author[
   addressref={aff2},                   % id's of addresses, e.g. {aff1,aff2}
   email={julius.parulek@uib.no}   % email address
]{\inits{JP}\fnm{J\'{u}lius} \snm{Parulek}}
\author[
   addressref={aff2},                   % id's of addresses, e.g. {aff1,aff2}
   email={Helwig.Hauser@uib.no}   % email address
]{\inits{HH}\fnm{Helwig} \snm{Hauser}}
\author[
   addressref={aff1},                   % id's of addresses, e.g. {aff1,aff2}
   email={kozlikova@fi.muni.cz}   % email address
]{\inits{BK}\fnm{Barbora} \snm{Kozl\'{i}kov\'{a}}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter the authors' addresses here        %%
%%                                          %%
%% Repeat \address commands as much as      %%
%% required.                                %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\address[id=aff1]{%                           % unique id
  \orgname{Masaryk University}, % university, etc
  %\postcode{15260}                                % post or zip code
  \city{Brno},                              % city
  \cny{Czech Republic}                                    % country
}
\address[id=aff2]{%                           % unique id
  \orgname{University of Bergen}, % university, etc
%  \street{210 South Bouquet},                     %
  %\postcode{15260}                                % post or zip code
  %\city{Williamsburg},                              % city
  \cny{Norway}                                    % country
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% Enter short notes here                   %%
%%                                          %%
%% Short notes will be after addresses      %%
%% on first page.                           %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{artnotes}
%\note{Sample of title note}     % note to the article
%\note[id=n1]{Equal contributor} % note, connected to author
%\note[id=n2]{Equal contributor} % note, connected to author
%\note[id=n3]{Equal contributor} % note, connected to author
%\note[id=n4]{Project leader and equal contributor} % note, connected to author
\end{artnotes}

\end{fmbox}% comment this for two column layout

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Abstract begins here                 %%
%%                                          %%
%% Please refer to the Instructions for     %%
%% authors on http://www.biomedcentral.com  %%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{abstractbox}

\begin{abstract} % abstract, must be under 350 words
%\parttitle{First part title} %if any
%Text for this section.

\parttitle{Background} Protein structures and their interaction with ligands have been in the focus of biochemistry and structural biology
research for decades. The transportation of ligand into the protein active site is often complex process, driven by
geometric and physico-chemical properties, which renders the ligand path full of jitter and impasses. 
This prevents understanding of the ligand transportation and reasoning behind its behavior along the path.

\parttitle{Results} To address the domain expert needs we design an explorative visualization solution based on a multi-scale simplification model. It helps to navigate the user to the most interesting parts of the ligand trajectory by exploring different parameters of the ligand and its movement, such as its distance to the active site, changes of amino acids lining the ligand, or ligand "stuckness".
The process is supported by three linked views -- 3D representation of the simplified trajectory, scatterplot matrix, and bar charts with line representation of ligand-lining amino acids.

\parttitle{Conclusions}  The usage of our tool is demonstrated on simulation of molecular dynamics simulation generated by the domain experts. The tool was tested by the domain experts from protein engineering and the results confirm that it helps to navigate the user to the most interesting parts of the ligand trajectory and to understand the ligand behavior. 


%\parttitle{Second part title} %if any
%Text for this section.
\end{abstract}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The keywords begin here                  %%
%%                                          %%
%% Put each keyword in separate \kwd{}.     %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{keyword}
\kwd{Molecular Visualization}
\kwd{Bioinformatics Visualization}
\kwd{Computational Proteomics}
\end{keyword}

% MSC classifications codes, if any
%\begin{keyword}[class=AMS]
%\kwd[Primary ]{}
%\kwd{}
%\kwd[; secondary ]{}
%\end{keyword}

\end{abstractbox}
%
%\end{fmbox}% uncomment this for twcolumn layout

\end{frontmatter}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                                          %%
%% The Main Body begins here                %%
%%                                          %%
%% Please refer to the instructions for     %%
%% authors on:                              %%
%% http://www.biomedcentral.com/info/authors%%
%% and include the section headings         %%
%% accordingly for your article type.       %%
%%                                          %%
%% See the Results and Discussion section   %%
%% for details on how to create sub-sections%%
%%                                          %%
%% use \cite{...} to cite references        %%
%%  \cite{koon} and                         %%
%%  \cite{oreg,khar,zvai,xjon,schn,pond}    %%
%%  \nocite{smith,marg,hunn,advi,koha,mouse}%%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%% start of article main body
% <put your article body there>

%%%%%%%%%%%%%%%%
%% Background %%
%%
\section*{Background}
The study of reaction processes between different types of molecules has been an important research problem already for decades.
A proper understanding of the processes occurring when two or more molecules react helps in the design of new chemical matters, e.g., in drug design or protein engineering.
Here, the researchers aim to combine a protein with a given ligand in order to design a new drug or to change protein properties and their function.
In these particular cases the ligand has to be transported from the outer solvent to the protein active site where the chemical reaction between the ligand and the amino acids surrounding the active site takes place.
The consecutive reaction then changes the composition and properties of both molecules. 
In protein engineering, for example, the goal is to alter the protein properties so that the new protein is, e.g., more stable and resistant to organic cosolvents~\cite{Koudelakova2013}.

The design complexity of such reactions lies namely in the transportation of the ligand to the protein active site.
As the active site is usually buried deeply in the protein structure and thus inaccessible directly from its surface, the ligand has to find a suitable transportation path through the protein structure.
This process, called molecular docking, is very complex, lengthy, and its analysis is heavy on computational resources. 
Therefore, researchers aim at solutions that simplify and ease the analysis for proper ligand binding.
Currently available solutions often focus on detection of possible ligand transportation paths through the protein, called tunnels.
These solutions are mostly based on the geometric analysis, e.g., CAVER~\cite{Chovancova2012}, MOLE~\cite{Sehnal2013}, or MolAxis~\cite{Yaffe2008}.
Figure~\ref{fig:tunnel} shows an example of small ligand passing through a tunnel computed by CAVER algorithm and visualized using CAVER Analyst~\cite{Kozlikova2014}. 
The tunnel is colored with respect to the hydrophobicity of the surrounding amino acids and the geometric bottleneck of the tunnel is clearly visible.

\begin{figure}[htb]
	\centering
  \includegraphics[width=1\linewidth]{img/tunnel.png}
  \caption{\label{fig:tunnel} Small ligand (green) passing through a tunnel represented by the transparent surface. The surface encloses the available empty space along the tunnel centerline and is colored with respect to the hydrophobicity of the surrounding amino acids (hydrophobic -- red, hydrophylic -- blue, neutral -- violet). The active site is located in the left ending of the tunnel.}
\end{figure}

Other approaches, such as MoMA-LigPath~\cite{Devaurs}, aim at simulating the ligand transportation itself.
Nevertheless, simulating the ligand docking using current computational approaches is still a challenging problem.
There are several available variants of molecular simulation methods devised specifically to this problem. 
Among these, Steered Molecular Dynamics~\cite{Isralewitz} and Random Acceleration Molecular Dynamics~\cite{Ludemann} are able to simulate the ligand binding. 
Such simulations produce a large amount of data containing the ligand movements that are to be explored by the domain expert.
As the length of these simulations reaches often hundreds of thousands of time steps, it becomes impossible for the domain experts to visualize and observe the ligand movement in a frame-by-frame manner.
Moreover, the simulation often contains movements that are irrelevant to the ligand binding.
For example, a significant portion of the simulation the ligand usually spends outside the protein searching for a proper tunnel gorge to enter the molecule. 
It also often happens that the ligand enters the molecule via a wrong tunnel and is then evicted from the molecule and searches for the entrance repeatedly.
Therefore, the "true" active site entering can be in fact captured in a smaller subset of the original sequence.

In this paper, we propose a new visual analysis system that addresses the aforementioned challenges, i.e., the transportation of a ligand to the active site. 
We aim to provide the domain experts with a tool for intuitive and interactive exploration of already captured molecular dynamics simulation containing the process of ligand binding. 
Using our proposed solution the users are able to distinguish between the parts of the simulation where the ligand searches for the proper path to the active site (searches the tunnel gorge or enters the wrong tunnel) and the part where the ligand finally leads to and reaches the active site.
Furthermore, our solution introduces a method for simplification of the original scattered ligand trajectory.
We suggest the automatic simplification of the trajectory which can be further manually adjusted, i.e., selected parts of the trajectory can be further simplified.
So the trajectory can be simplified in a hierarchical manner and further interactively explored using the other proposed views.
One of these views is the 3D visualization where the simplified trajectory can be displayed in the spatial context of the whole molecule.
The trajectory can be color-coded according to different criteria and selections performed in the other views are highlighted by a thick tube.
Another available view is the scatterplot matrix which helps to explore different properties of the ligand along its trajectory, e.g., the direction or distance to the active site. 
Selections performed in these scatterplots are highlighted not only in the 3D view but also in another proposed view, the bar chart representing the temporal changes of selected ligand properties. 
This view enables to aggregate the time steps to better reveal the trends in ligand behavior.
The bar chart is equipped with the line representation of amino acids lining the ligand along its trajectory. 
This helps the user to reveal the closest amino acids directly influencing the ligand behavior in different time steps.

Our proposed solution was thoroughly tested and evaluated by the domain experts from protein engineering area.
The evaluation was performed on several simulations captured and analyzed before using the traditional approach (manual exploration of ligand in individual time steps). 
Therefore, we also describe the process of using our solution when exploring these molecular dynamics simulations and summarize the feedback from the domain experts.

\subsection*{Related Work}
The interactive exploration of ligand paths leading through protein structures is a complex problem that requires an elaborate analysis of existing and convenient approaches.
As our system addresses different areas, this section will be divided with respect to these areas as well. 
Studying the ligand transportation path is tightly related to the analysis of its trajectory.
Thus, the first part will cover the topic related to the trajectory analysis, simplification, and visualization.
Then the existing representations of tunnels and their surrounding amino acids will be addressed.

\subsubsection*{Trajectory Analysis, Simplification, and Visualization}
When searching for the most appropriate trajectory simplification method, the taxonomy of different movement patterns introduced by Dodge et al.~\cite{Dodge2008} can help to categorize the type of motion and to propose an appropriate solution for the analysis and simplification.
Then, Andrienko et al.~\cite{andrienko2013} describe a legacy of tools and approaches to analyze trajectory data.
With respect to the trajectory simplification, they introduce two approaches to data abstraction representing a necessary step to achieve the reduction of the data in order to achieve informative visualizations.
The first approach is characterized by omitting unnecessary positions and segments, while the second exploits data subsampling.
On the other hand, they state that there is no general method to sample trajectories.
Thus the suitability of particular sampling method is deduced from what information is considered important.
A recent work of Vrotsou et al.~\cite{Vrotsou2015} introduce a systematic stepwise methodology for trajectory simplification with emphasis on visual analysis.
Even though they primarily developed the tool to analyze three dimensional trajectories, it is not limited to them only.

In the field of molecular biology as well as systems biology, there are several examples of methods focusing on trajectory analysis.
Bidmon et al.~\cite{bidmon2008} present an abstract way of identification and visualization of solvent molecules' pathways within molecular dynamics. 
In comparison with the previous solutions, their approach preserves valuable information on the directions and velocities of water molecules routing along these paths. 
Another approach that describes a guidance through a complex simulation trajectories in systems biology is presented by Luboschik et al.~\cite{Luboschik2012}. 
The method addresses biochemical reaction networks and aims to provide the users with a tool for investigating the overall behavior of a modeled system and detailed behavior at the same time.

\subsubsection*{Visualization of Cavity and Tunnel Features}
Phillips et al.~\cite{phillips2010} propose a method to quantitatively estimate molecular features, e.g., volume and surface areas, via a ray-casting technique. 
This involves the computation of cavities.
Lindow et al.~\cite{LindowBH11} introduce a technique that allows to extract significant paths from the molecules. 
In their approach the authors utilize a Voronoi diagram of spheres. 
Their final visualization is achieved by means of placing light sources on the extracted paths to enhance the presence of tunnels.
Parulek et al.~\cite{parulek2012} exploit scatterplots to communicate the evolution of protein voids. 
In their later study~\cite{parulek2013} they also suggest to utilize amino acids physico-chemical properties related to cavities to help users to navigate through their occurrences. 
The visualization is then achieved in the focus and context manner.
Lindow et al. \cite{Lindow2013} present an approach for visualization of temporal evolution of cavities in a temporal graph, which, to a certain extent, resembles our temporal tunnel model. 
In this work they focus on the interactive exploration of the dynamics of protein cavities that can form the transportation path for a ligand. 
They calculate and visualize the cavity volume and analyze the time-dependent changes of the cavity structure.
The cavity dynamics is captured by rendering the cavities in a single image. 
The final visualization is achieved through the molecular surface representation colored according to time. 
Krone et al.~\cite{krone2013features} present a similar approach where they extract and track tunnels in MD simulations. 
They exploit temporal graphs to communicate the evolution of surface areas of tunnels. 
In our technique, we go one step further, where in addition, we incorporate the ligand-protein interaction information and provide users with the means to navigate through details of this interaction. 
Moreover, we analyze sequences of several thousands of frames, which is not the case in aforementioned approaches.
Kozl\'{i}kov\'{a} et al.~\cite{kozlikova2014vcbm} propose a way to seamlessly visualize the geometry and shape of tunnels across MD simulation. 
Here, they aim at 3D visualization solely, which is not suitable way of exploring and understanding of thousands of simulation time-steps.
By\v{s}ka et al.~\cite{byska2016} introduce an approach to interactively explore the tunnel objects in MD simulations. 
Similarly to our proposed approach, they visualize the time-varying tunnel as a profile graph that includes information on surrounding amino acids. 
Nevertheless, they focus on a single tunnel instance and do not provide the means to explore the ligand-protein interaction.

%-------------------------------------------------------------------------
\section*{Problem Description and Input Data}
When studying long molecular dynamics simulations, the researchers have to face namely the following high-level tasks:
\begin{itemize}
\item Detect the part of the simulation where the ligand enters the protein and finally reaches the active site.
\item Explore this route in detail and detect its bottlenecks.
\end{itemize}
In consequence, these steps are performed in order to reveal the parts of the trajectory where the ligand gets stuck.
It means that in such parts there are some obstacles made by the surrounding amino acids.
These obstacles can be geometric (the empty space between these amino acids is too narrow) or physico-chemical (the properties of the amino acids are incompatible with the ligand properties), or their combination.
The geometric obstacles can be detected by using an algorithm for tunnel computation in molecular dynamics (e.g., CAVER~\cite{Chovancova2012} or MOLE~\cite{Sehnal2013}).
These tools are able to produce the information about time evolution of individual tunnels and for each tunnel detect its bottleneck -- the narrowest part limiting the size of the ligand aiming to pass through this tunnel.

However, this bottleneck can be to some extent influenced by the passing ligand and this influence is also determined by the physico-chemical properties of the ligand and the bottleneck-lining amino acids (e.g., hydrophobicity or partial charges of individual atoms).
To reveal such dependencies, it requires the involvement and experience of the domain expert who has to study the ligand trajectory captured in the molecular dynamics simulation.
%And for such task a proper visual representation and guidance through such large data is substantial.

In protein engineering, all these efforts can finally lead to the detection of amino acids along the ligand trajectory which caused some problems, i.e., were the key players in situations when the ligand got stuck.
Such amino acids are then the best candidates for subsequent mutation of the protein chain when these amino acids are replaced by more suitable ones wrt. their size and properties.
On the other hand, in drug design the aim is to propose modifications of the ligand to increase the binding likelihood in order to design a better drug produced from this ligand.

The input data is obtained from the simulations of molecular dynamics which contains the movements of the protein and one or more ligands.
Ligands can follow different routes -- they can be transported from the outside environment to the protein active site or vice versa (after the desired reaction the product leaves the protein).

The length of the simulations may vary from few hundreds to hundreds of thousands of time steps.  
This depends namely on the ligand velocity and its ability to find the proper path to the active site.

\begin{figure}[b!]
	\centering
  \includegraphics[width=0.95\linewidth]{img/lig_movement.png}
  \caption{\label{fig:lig_movement} Ligand trajectory (gray) for a simulation containing $50.000$ time steps. The protein chain is depicted in blue.}
\end{figure}

Long simulations can often contain many time steps when the ligand followed an impasse -- it tried to find the proper entrance point to the protein (tunnel gorge) or entered a wrong tunnel and had to return.
Such parts of the simulation could be of less interest so that the user should be provided with a possibility to filter them out and focus only on the interesting parts.
Figure \ref{fig:lig_movement} shows such example when the ligand trajectory consists of $50.000$ time steps. 
It is visible that in most of the time the ligand travels around the protein chain (depicted in blue).
Therefore, only a fraction of the trajectory captures its transportation to the active site.


\section*{System Overview}
Our proposed solution consists of several linked views covering individual steps of the exploration workflow (see Figure~\ref{fig:overview}). 
In the first part, the user is provided with the overview representation of the whole loaded trajectory (Figure~\ref{fig:overview} 1). 
This simple 1D representation encodes different routes of the ligand -- green color highlights the parts of the trajectory where the ligand is inside the protein, white and yellow colors encode the situation when the ligand is outside the protein or on it surface respectively.
Blue color shows the time portions when the ligand was close to the active site.
By selecting a part of the trajectory using this representation, the user is automatically navigated to this selected part and all following representations show only this part.

\begin{figure*}[thb]
	\centering
  \includegraphics[width=0.95\textwidth]{img/overview.png}
  \caption{\label{fig:overview} Overview of the proposed system. 1 -- overview of the whole loaded trajectory, 2 -- scatterplot matrix plotting different attributes of the ligand trajectory, 3 -- bar charts and line representation showing ligand-lining amino acids, 4 -- spatial representation of the trajectory.}
\end{figure*}

Among these views belong the scatterplot matrix (Figure~\ref{fig:overview} 2) enabling to explore different attributes of the ligand and its trajectory.
These attributes are described in detail in section Derivation of Attributes.
By interactive manipulation with these matrices the user can select a subset of time steps in which the ligand behaved in a desired way.
The selection is highly dependent on the current tasks the user aims to perform.
For example, when the user searches for the parts of the simulation where the ligand got stuck for a certain amount of time steps and at the same time was close to the active site, he or she plots and interacts with the matrix showing the relationship between the ligand stuckness and its distance from the active site.

The selected time steps can be further explored using the combination of bar chart and line representations (Figure~\ref{fig:overview} 3). 
The bar chart enables to follow a selected attribute over time.
Furthermore, individual bars can represent an aggregated information for selected portions of time steps.
The aggregation is performed on uniformly divided time intervals.
Then they show the average values of selected attributes.
The line representation informs about the amino acids closest to the ligand along its path.
This information helps to determine those amino acids which play a substantial role in the ligand transportation.

These views are linked with the three-dimensional view (Figure~\ref{fig:overview} 4) where the user can observe the ligand trajectory in the context of the protein structure.
As the original trajectory is highly scattered, we propose a simplification algorithm which aims to reveal the trends in the ligand movement and suppress small insignificant movements.
The algorithm is described in detail in the Trajectory Simplification section.


\subsection*{Trajectory Simplification}
\input{parts/simplification}

\subsection*{Derivation of Attributes}
In the previous section we described the simplification of the ligand trajectory. 
On one hand, such approximation helps to understand the overall ligand movements. 
On the other hand, it also suppresses vast amount of details that are important for complete understanding of the ligand movements inside protein.

In order to preserve this information and hence allow the biochemist to explore the ligand behavior in detail we evaluate interesting geometric and physico-chemical attributes of the ligand trajectory on multiple levels. 
These attributes are then communicated to the user in several ways (using the scatterplot matrix, the bar charts, and different coloring of the trajectory in the 3D view).

Based on several discussions with the domain experts, we detected the following attributes as the most significant: "stuckness" of the ligand in one place, its distance from the active site, the direction of its movement, the amount of surrounding free space, changes of surrounding amino acids and their properties, the hydrophobicity and charge profiles of amino acids and atoms around the ligand, and the speed of the ligand along the trajectory. 
Figure~\ref{fig:attributes} aims to illustrate the individual attributes.

The remaining part of this section describes these attributes in detail, along with the process of deriving these attributes from the original data, and rationalizes their usefulness.
    
\begin{figure*}[thb]
	\centering
  \includegraphics[width=0.95\textwidth]{img/attributes.png}
  \caption{\label{fig:attributes} Illustration of the attributes derived for the ligand and its trajectory. AA1-AA4 denote amino acids lining the ligand in a specific position.}
\end{figure*}

\paragraph{\textbf{Stuckness}} is important attribute which comes from the fact that if the ligand got stuck in some place, it means that there was some obstacle (geometric, physico-chemical, or combined) which prevented the ligand to pass towards the active site.
The biochemists are highly interested in these obstacles and want to explore them in more detail in order to reveal the reason of this problem and possibly propose a replacement (mutation) of some amino acids surrounding this problematic spot.
We consider the ligand to be stuck if it oscillates around a specific place in the protein. %(w.r.t. the active site)
To estimate the stuckness, we measure how much was the movement of the ligand straight, i.e., whether it moved significantly in some direction or its movement was rather random.
The stuckness is estimated in a neighborhood of $2n + 1$ consecutive ligand positions (w.r.t. time).
The size of the neighborhood $n$ can specified by the user.
For a set of ligand positions $P$ we define the stuckness as:
\begin{equation}
  s(x) = \frac{\max_{\,-n \le i, j \le n} \{ d(P(x + i), P(x + j)) \}}{\sum_{-n \le i < n}{d(P(x + i), P(x + i + 1))}},
  \label{eq:straightness}
\end{equation}
where $P(x)$ is position of the ligand in snapshot $x$ and $d(x, y)$ is euclidean distance of $x, y \in \mathbb{R}^3$.
To be complete, in the beginning (or in the end) of the trajectory, we shrink the neighborhood from left (or right).
Furthermore, we set $n = 4$ by default as this value was obtained experimentally.

\paragraph{\textbf{Distance}} of the ligand from the active site provides the biochemists with the information whether some observed behavior occurred in a vicinity of the active site, the surface, or somewhere between. 
This can be very helpful for instance when we want to mutate some amino acids along the trajectory in order to remove the stuckness of the ligand in that area. 
Therefore, this attribute is naturally connected with the ligand stuckness.
In these cases it is essential to see also where exactly the unwanted behavior of ligand happened.
In order to evaluate the current distance of the ligand from the active site, we first extract the position of the active site $A(x)$ in each time step $x$. 
Here the active site is defined by a set of surrounding amino acids and we compute the position of the active site as the center of mass of these amino acids, i.e., their atoms. 
Once $A(x)$ is determined, we compute distance of the ligand position $P(x)$ to the active site as $d_{AS}(x) = d(P(x), A(x))$ where $d(x, y)$ is again euclidean distance of $x, y \in \mathbb{R}^3$.

\paragraph{\textbf{Direction}} of the ligand movement with respect to the active site is another essential attribute. 
During the MD simulation the ligand can enter and exit the protein tunnel repeatedly. 
Therefore, biochemists want to distinguish between intervals where the ligand was traveling towards the active site or in the opposite direction (i.e., towards the outer solvent). 
To compute the direction we employ the \textbf{distance} attribute such that we evaluate whether the distance increases between two subsequent time steps, when moving towards the active site, or decreases, when moving in the opposite direction.

\paragraph{\textbf{Amount of free space}} surrounding the ligand can help the biochemists to understand why the ligand in some places got stuck.
In such cases there was not enough free space for the passage of the ligand or the ligand was repelled by some amino acid with incompatible physico-chemical properties.
the amount of free space around the ligand can be obtained in two ways.
When the user has the information about tunnels (e.g., calculated by CAVER tool~\cite{Chovancova2012}), we can use it for the assessment of the free space around the ligand.
We take the tunnel in the corresponding time step and search for its sphere in which the ligand is positioned. 
The radius of this sphere is then taken as the descriptor of the free space.
However, when the information about the tunnel is not available, we compute the free space in the following way.
We denote this approach as the detection of temporal tunnel.
The temporal tunnel is defined by a set of spheres where the number of spheres corresponds to the number of time steps in the simulation.
Each sphere is taken from one time step and it defines the maximum empty space surrounding the ligand.
For each time-step we firstly retrieve ten protein atoms closest to the ligand. 
Their mean position defines the center of the temporal tunnel sphere while the radius is determined as the minimum distance from this center to the closest protein atom.

\paragraph{\textbf{Changes of amino acids and their properties}} help to understand when the ligand was in contact with a specific set of amino acids and when this set changed.
In other words, the changes of amino acids can reveal those parts of the trajectory where the ligand significantly changed its position with respect to the protein. 
The higher number of different amino acids surrounding the ligand between two consecutive time steps corresponds to variations inside the protein tunnel as well.
Moreover, whether the ligand will be able to pass through the tunnel and reach the active site is highly dependent not only on the geometric properties of the tunnel but also on the physico-chemical properties of the surrounding amino acids and their atoms (e.g., hydrophobicity or electric charge).
This attribute is tightly connected with the ligand stuckness and subsequent proposal of candidates for amino acid mutations.

\paragraph{\textbf{Hydrophobicity profile}} is related to the amino acids surrounding the ligand trajectory.
It aims to add the user the information about physico-chemical property -- hydrophobicity -- of these amino acids. 
This is very useful in cases when the ligand got stuck and this was not caused by any geometric obstacle.
This attribute is calculated as the average value from all ligand-lining amino acids when the lining distance is taken as 2 \AA.

\paragraph{\textbf{Charge profile}} has a very similar purpose as the hydrophobicity profile.
It conveys the information about another significant physico-chemical parameter, atom partial charge, calculated for each atom in the distance of 2 \AA\ from the ligand and averaged.

\paragraph{\textbf{Speed}} can be easily described and estimated in relative terms.
%is one of the most abstract attributes. 
%We consider the ligand to be stuck if it oscillates around a specific place in the protein (w.r.t. the active site).  
%In other words, we evaluate the time for which the ligand remained close to a single spot inside the protein. 
Since the original data are uniformly sampled (i.e., we have the information about the exact ligand position in every two femtoseconds), we can measure the distance between two subsequent time steps and thus obtain the speed of the ligand in that particular area.
%The higher the speed, the higher the distance the ligand can travel and the less it got stuck.

\section*{Visual Exploration}
In this section we describe the individual views supported by our system.
All views are interactively linked, i.e., the user can select a specific parts of the ligand trajectory and these parts are highlighted also in the remaining views.

\subsubsection*{Trajectory Overview}
This simple 1D representation helps to navigate the user to the interesting parts of the simulation.
It is depicted as a bar colored according to the ligand position with respect to the protein surface and active site.
We assort the ligand movement into four categories: 
\begin{itemize}
\item \textbf{Outside} -- the ligand is moving outside of protein.
\item \textbf{Surface} -- the ligand is moving alongside the protein surface.
\item \textbf{Inside} -- the ligand is moving within the protein.
\item \textbf{Active Site} -- the ligand is moving in close proximity to the active site.
\end{itemize}
This representation enables quick filtering of irrelevant data (e.g., parts of the simulation when the ligand is moving outside the protein), as well as navigation to the potentially important parts.
This is reached by using a brushing tool that allows to draw a rectangle over the desired part of the simulation and the corresponding time steps are selected and used for further exploration.
The selected time interval is then displayed in the bar chart as well as in the scatterplot matrix.

\subsubsection*{Bar Chart and Line View}
This view provides a closer look at the ligand movement with respect to the properties of the temporal tunnel.
It consists of two parts -- the bar chart and line representation of all amino acids lining the ligand trajectory.

By default, each bar in the bar chart represents one time step of the simulation, though if needed (e.g., if we select long time sequence) the neighboring bars can be uniformly aggregated.
The bars are selectable and the selected time steps are highlighted in the scatterplot matrix as well as in the 3D View.
The height of the bar is set to represent one of the geometric or physico-chemical attributes described previously.
The color of selected bars corresponds to the coloring of the trajectory overview, indicating the ligand position at a given time step.
The user can visualize only the selected bars.
When the set of selected bars is not continuous, the user is informed about it by a small wave inserted between the bars.

The information about the surrounding acids is also very crucial when exploring the sites when the ligand got stuck. 
For each time step within the temporal block we compute the three closest amino acids from the ligand. 
We provide the visual representation of this list (Figure ~\ref{fig:aacids}), where each amino acid defines one line in the list. 
The line is colored with respect to a selected physico-chemical property -- our representation supports switching between hydrophobicity, partial charges, and donors and acceptors.
The interruption in some of these lines is caused by the fact that the given amino acid is not present around the ligand in the corresponding temporal block.

\begin{figure}[htb]
	\centering
  \includegraphics[width=0.95\linewidth]{img/aacids.png}
  \caption{\label{fig:aacids} Representation of amino acids surrounding the temporal tunnel and their physico-chemical properties. Coloring corresponds to a selected property, the line interruption means that this amino acid was not detected around the ligand in the corresponding temporal block.}
\end{figure}

\subsubsection*{Scatterplot Matrix}
For detailed analysis of geometric and physico-chemical attributes of the ligand trajectory at given time steps we propose the scatterplot matrix.
The axes of one scatterplot represent a pair of user-selected attributes.
Each point in the scatterplot then represents the values of these attributes at one time step.
As a result, the scatterplot can easily reveal trends and relationships between attributes.
Interactive manipulation with the scatterplot, such as selection, zooming, and change of displayed attributes provides an easy way for manual data filtering. 
To eliminate the noise in the data caused by ligand jittering, we propose the \textit{sliding window} smoothing function.
The sliding window function assigns to each time step the averaged value from its neighborhood.
The size of the neighborhood is again regulated by the user. 

The plots can further be stacked, forming a matrix and thus showing relationship between multiple attributes at once.
The stacked plots are also interactively interconnected and the selection in Scatterplot Matrix is interactively linked with Barchart View and 3D View.

Figure \ref{fig:scatterplot} shows the scatterplot matrix for a sequence of $500$ time steps.

\begin{figure}[htb]
	\centering
  \includegraphics[width=0.95\linewidth]{img/scatterplot.png}
  \caption{\label{fig:scatterplot} Scatterplot matrix depicting time vs. distance and time vs speed property before (up) and after (down) smoothing by the sliding window function.}
\end{figure}

\subsubsection*{3D View}
This view provides an overview of the ligand trajectory in 3D.
However, as was shown in Figure \ref{fig:lig_movement}, the ligand trajectory contains a lot of noise, due to the jittery movement of the ligand.
Therefore, we introduced the previously described trajectory simplification.
Using this function the whole trajectory can be automatically simplified.
If the automatic simplification does not lead to satisfactory results, selected parts of the trajectory can be further simplified manually.
The trajectory is colored according to one of the geometric and physico-chemical attributes. 

\section*{Analysis Procedure}

\begin{figure*}[htb]
	\centering
  \includegraphics[width=0.95\textwidth]{img/case_overview.pdf}
  \caption{\label{fig:case_overview} Overview of the simulation consisting of $50.000$ time steps. Selected part was explored in detail in our case study.}
\end{figure*}

In this section we will describe one of case studies which we have performed using our novel exploration tool.
In this particular study the biochemists explored molecular simulation of Haloalkane dehalogenase together with the ligand. 
Since the simulation consists of $50.000$ time steps, it would take enormous amount of time to analyze it using only common techniques such as 3D animation.
In this section we will show how this can be significantly improved using our exploration tool.
After the precomputation stage the user can immediately start to explore individual views described in the previous sections.

We start with the rough analysis of the overall behavior of the ligand using our simulation overview (see Figure~\ref{fig:case_overview}).
Utilizing this view the biochemist can immediately see that the whole simulation contains three main (two long and one relatively short) parts where the ligand was inside the protein (blue and green colors).
These parts are divided by two orange and white blocks signalizing that the ligand was moving near to the protein surface (orange) or even entirely left the protein (white).
 
The mentioned color mapping on the simulation overview suggests that in this particular simulation the ligand started in the active site, then traveled towards the protein surface, and finally left the molecule.
After spending some time outside it found again its way into a protein tunnel and traveled back to the active side.
The whole process is then repeated again while this time the ligand spent significantly more time outside the protein or nearby its surface and returned just for last few hundreds of time steps. 

After the first evaluation of the overall ligand trajectory the biochemists stated that they would like to evaluate the interactions between the ligand and protein in more detail using the following scenario (see Figure~\ref{fig:case_overview}). 
They focused on the part where the ligand traveled from the active site to the protein surface (see Section~\ref{sec:case1}). 

The exploration of this part of the simulation allows the biochemists to closely evaluate the influence of the protein tunnel and its properties to the ligand movements.

The biochemists selected the first part of the simulation consisting from circa $17.000$ time steps (see Figure~\ref{fig:case_overview}). 
From this point the biochemists continued to more detailed exploration using brushing and linking in other three views -- namely using the 3D, scatterplot, and bar chart views (see Figure~\ref{fig:case_hydro_distance} -- top).
Each of these views provides slightly different representation of the data thus allowed the biochemist to perform different operations.

For instance, by using the scatterplot set to show the ligand distance from the active site over time (see Figure~\ref{fig:case_hydro_distance} -- top) we immediately revealed that the ligand was actually near to the protein surface (represented by high peaks) in few additional times during this part of the simulation.
The same behavior can be actually observed also in the simulation overview (see Figure~\ref{fig:case_overview}) where these cases are highlighted by slim orange lines. 
But these cases are much better visible in the scatterplot view.

Another interesting behavior which can be seen with this setup is that the ligand repeatedly traveled back and forward from the active site (lower values) to the surface (higher values) in the first (and again in the last) part of the selected interval.
This behavior can suggests that the ligand had actually problems to bind with the protein in the active site.

 
 
%{\color{red}While there is a huge part (in the middle) where the ligand is actually stuck oscillating nearby the active site.}

%Moreover, the scatter plot view enables also to select these time steps (or the opposite) for future exploration in much more convenient way. 

When evaluating the reactivity between the ligand and protein active site it is important to determine the amino acids that interact with the ligand. 
When these amino acids are non-polar then the ligand has to be non-polar as well.
Similarly, polar amino acids attract the polar ligand.
Therefore, the biochemist were interested in seeing whether there are some polar amino acids inside the active site.
The polarity of amino acids is closely related to their hydrophobicity -- hydrophobic amino acids are non-polar while hydrophilic amino acids are polar. 

The desired information can be retrieved using our exploration tool in many ways.
Here we show one of the possible solutions used by our domain experts in the real test scenario.

The biochemists first added another scatterplot showing the hydrophobicity profile along the ligand trajectory (see Figure~\ref{fig:case_hydro_distance}).
Since both scatterplots are interactively connected in real-time, the biochemists could select the low hydrophobicity values and immediately observed at which distance they are positioned with respect to the ligand trajectory. 

\begin{figure}[htb]
	\centering
  \includegraphics[width=0.95\linewidth]{img/case_hydro_distance.png}
  \caption{\label{fig:case_hydro_distance} {Top -- the distance from the active site over time. Bottom -- hydrophobicity profile of the trajectory.}}
\end{figure} 

The results confirmed the behavior common for most of the proteins, namely the fact that the part of the trajectory in greater distance from the active site (i.e., nearby the surface) is surrounded mostly by hydrophilic amino acids.
On the other hand, when the ligand reaches the inner part of the protein, it is surrounded by highly hydrophobic amino acids.

However, as can be seen in the bottom part of Figure~\ref{fig:case_hydro_distance}, there are few amino acids that are actually hydrophilic but located in the close vicinity to the active site. 
In order to easily pinpoint these specific amino acids, the biochemists set the scatterplot to visualize the exact relationship between the distance and hydrophobicity (see Figure~\ref{fig:case_hydro_distance_2} -- bottom).
By creating the selection in this view, the biochemists were able to easily pinpoint the part of the simulation closer to the active site which contained the hydrophilic amino acids (see Figure~\ref{fig:case_hydro_distance_2} -- top).

\begin{figure}[htb]
	\centering
  \includegraphics[width=0.95\linewidth]{img/case_hydro_distance_2.png}
  \caption{\label{fig:case_hydro_distance_2} {\color{red}TODO}}
\end{figure}

\begin{figure}[htb]
	\centering
  \includegraphics[width=0.95\linewidth]{img/case_hydro_distance_2_orig.png}
  \caption{\label{fig:case_hydro_distance_3} {\color{red}TODO} mam jet aj tenhle ale nevm zda ho pout -- to je ta prvni selekce?}
\end{figure}

As the next step, the biochemists were interested in seeing which amino acids are actually responsible for the described behavior.
For this reason they utilized the bar chart view containing the amino acid graph (see Figure~\ref{case_hydro_vs_dist_bar}).

\begin{figure}[htb]
	\centering
  \includegraphics[width=0.95\linewidth]{img/case_hydro_vs_dist_bar.png}
  \caption{\label{fig:case_hydro_vs_dist_bar} {\color{red}TODO}}
\end{figure}


In order to easily see the required data they limited the view only to selected time steps and set high level of aggregation (each bar contained about 20 time steps).
Even though the amino acid view contained the list of all amino acids surrounding the ligand in given time steps, it was relatively easy to spot the hydrophobic ones.
This was enabled thanks to the ability of the view to color individual amino acids based on their properties. 
Here the biochemists selected the green amino acids (asparagine with ID 38 and aspartic acid with ID 103). 
As can be seen in Figure~\ref{fig:case_hydro_vs_dist_3d} the selected amino acids exactly correspond to the estimated active site in the protein.

\begin{figure}[htb]
	\centering
  \includegraphics[width=0.95\linewidth]{img/case_hydro_vs_dist_3d.png}
  \caption{\label{fig:case_hydro_vs_dist_3d} {\color{red}TODO} ne se zepta, SS jsou dodrban, musm opravit :D}
\end{figure}

In the last step the biochemists went back to the bar chart, lowered the aggregation level and selected all bar charts containing the two hydrophobic amino acids (see Figure~\ref{fig:case_hydro_vs_dist_residues_selected}). 
{\color{red}} I thought/hoped, that this would clearly show, that these two AA are involved in the "not being able to bind" section (\ie where the high peaks are) and hence prove my theory stated at the begining, but this clearly have not  happen so, what next?.

\begin{figure}[htb]
	\centering
  \includegraphics[width=0.95\linewidth]{img/case_hydro_vs_dist_residues_selected.png}
  \caption{\label{fig:case_hydro_vs_dist_residues_selected} {\color{red}TODO}}
\end{figure}


%This allows them to closely evaluate the influence of the protein tunnel and its properties to the ligand movements. 

%Exploration of this part of the simulation may help to reveal the essential conditions needed for this particular ligand to enter the protein.



%{\color{red}Note that the biochemists can also estimate that the provided data were probably computed using Steered molecular dynamics (SMD) \todo{popsat co je SMD in RW?} simulation since the simulation starts with ligand in the active site (signalized by red color.)}
 
\section*{Results and Discussion - BK}
use case + feedback

\section*{Conclusions - All}
and future work 

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
%%                                          %%
%% Backmatter begins here                   %%
%%                                          %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{backmatter}

\section*{Competing interests}
The authors declare that they have no competing interests.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                  The Bibliography                       %%
%%                                                         %%
%%  Bmc_mathpys.bst  will be used to                       %%
%%  create a .BBL file for submission.                     %%
%%  After submission of the .TEX file,                     %%
%%  you will be prompted to submit your .BBL file.         %%
%%                                                         %%
%%                                                         %%
%%  Note that the displayed Bibliography will not          %%
%%  necessarily be rendered by Latex exactly as specified  %%
%%  in the online Instructions for Authors.                %%
%%                                                         %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% if your bibliography is in bibtex format, use those commands:
\bibliographystyle{bmc-mathphys} % Style BST file
\bibliography{bmc_article}      % Bibliography file (usually '*.bib' )

% or include bibliography directly:
% \begin{thebibliography}
% \bibitem{b1}
% \end{thebibliography}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Figures                       %%
%%                               %%
%% NB: this is for captions and  %%
%% Titles. All graphics must be  %%
%% submitted separately and NOT  %%
%% included in the Tex document  %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%
%% Do not use \listoffigures as most will included as separate files

%\section*{Figures}
%  \begin{figure}[h!]
 % \caption{\csentence{Sample figure title.}
  %    A short description of the figure content
   %   should go here.}
   %   \end{figure}

%\begin{figure}[h!]
 % \caption{\csentence{Sample figure title.}
  %    Figure legend text.}
  %    \end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Tables                        %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Use of \listoftables is discouraged.
%%
%\section*{Tables}
%\begin{table}[h!]
%\caption{Sample table title. This is where the description of the table should go.}
 %     \begin{tabular}{cccc}
%        \hline
 %          & B1  &B2   & B3\\ \hline
 %       A1 & 0.1 & 0.2 & 0.3\\
 %       A2 & ... & ..  & .\\
 %       A3 & ..  & .   & .\\ \hline
 %     \end{tabular}
%\end{table}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%                               %%
%% Additional Files              %%
%%                               %%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%\section*{Additional Files}
%  \subsection*{Additional file 1 --- Sample additional file title}
%    Additional file descriptions text (including details of how to
%    view the file, if it is in a non-standard format or the file extension).  This might
%    refer to a multi-page table or a figure.

%  \subsection*{Additional file 2 --- Sample additional file title}
%    Additional file descriptions text.

\end{backmatter}
\end{document}
